MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -2,11 +2,211 @@ import UIKit
+    private var correctAnswers = 0
+    //let currentQuestion = questions[currentQuestionIndex]
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        show(quiz: convert(model: questions[currentQuestionIndex]))
+    }
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(), // распаковываем картинку
+            question: model.text, // берём текст вопроса
+            questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)") // высчитываем номер вопроса
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        if isCorrect {
+            correctAnswers += 1
+        }
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        //imageView.layer.cornerRadius = 20
+        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { 
+            self.showNextQuestionOrResults()
+        }
+    }
+    
+    private func showNextQuestionOrResults() {
+        //вроде бы исправил этот баг ;) жду ответа в ревью!
+        buttonNo.isEnabled = true
+        buttonYes.isEnabled = true
+        if currentQuestionIndex == questions.count - 1 {
+            let text = "Ваш результат: \(correctAnswers) из 10"
+            let viewModel = QuizResultsViewModel(
+                title: "Этот раунд окончен!",
+                text: text,
+                buttonText: "Сыграть ещё раз")
+            imageView.layer.borderWidth = 0
+            show(quiz: viewModel)
+        } else {
+            currentQuestionIndex += 1
+            let nextQuestion = questions[currentQuestionIndex]
+            let viewModel = convert(model: nextQuestion)
+            imageView.layer.borderWidth = 0
+            show(quiz: viewModel)
+        }
+    }
+    
+    private func show(quiz step: QuizStepViewModel) {
+        // здесь мы заполняем нашу картинку, текст и счётчик данными
+        imageView.image = step.image
+        textLabel.text = step.question
+        counterLabel.text = "\(step.questionNumber)"
     }
+    
+    private func show(quiz result: QuizResultsViewModel) {
+        // здесь мы показываем результат прохождения квиза
+        let alert = UIAlertController(
+            title: result.title,
+            message: result.text,
+            preferredStyle: .alert)
+        
+        let action = UIAlertAction(title: result.buttonText, style: .default) { _ in
+            self.currentQuestionIndex = 0
+            
+            // скидываем счётчик правильных ответов
+            self.correctAnswers = 0
+            
+            // заново показываем первый вопрос
+            let firstQuestion = self.questions[self.currentQuestionIndex]
+            let viewModel = self.convert(model: firstQuestion)
+            self.show(quiz: viewModel)
+        }
```

Comment:
В ближайших спринтах вам расскажут про слабый список захвата. Сейчас здесь сильно удерживается ссылка на контроллер, такой код может приводить к утечкам памяти. Предлагаю попробовать переписать верно: сначала добавить `[weak self]`, затем вам понадобится обращаться к методам через `self?.`, чтобы минимизировать такие вызовы – просто выносим внутренний код в отдельную функцию: 

```suggestion
        let action = UIAlertAction(title: result.buttonText, style: .default) { [weak self] _ in
           self?.restartQuiz()
        }
```